frontend/src/metabase-lib/v1/expressions/suggest.ts

```diff
@@ -269,6 +268,23 @@ export function suggest({
     }
   }
 
+  // add help text to the suggestion if possible
+  suggestions = suggestions.map(suggestion => {
+    if (!database || suggestion.type !== "functions") {
+      return suggestion;
+    }
+
+    const name = getMBQLName(suggestion.name);
+    if (!name) {
+      return suggestion;
+    }
+
+    return {
+      ...suggestion,
+      helpText: getHelpText(name, database, reportTimezone),
+    };
+  });
+
```

Comment:
I'd rather have us add `helpText: getHelpText(name, database, reportTimezone)` where we create the suggestion objects instead of having this extra bit of logic here.

```patch
diff --git a/frontend/src/metabase-lib/v1/expressions/suggest.ts b/frontend/src/metabase-lib/v1/expressions/suggest.ts
index 6098a45c77..9ff94219ce 100644
--- a/frontend/src/metabase-lib/v1/expressions/suggest.ts
+++ b/frontend/src/metabase-lib/v1/expressions/suggest.ts
@@ -120,6 +120,7 @@ export function suggest({
       index: targetOffset,
       icon: "function",
       order: 1,
+      helpText: getHelpText("case", database, reportTimezone),
     });
     suggestions.push(
       ...Array.from(EXPRESSION_FUNCTIONS)
@@ -134,6 +135,7 @@ export function suggest({
           index: targetOffset,
           icon: "function",
           order: 1,
+          helpText: getHelpText(func.displayName, database, reportTimezone),
         })),
     );
     if (startRule === "aggregation") {
@@ -268,23 +270,6 @@ export function suggest({
     }
   }
 
-  // add help text to the suggestion if possible
-  suggestions = suggestions.map(suggestion => {
-    if (!database || suggestion.type !== "functions") {
-      return suggestion;
-    }
-
-    const name = getMBQLName(suggestion.name);
-    if (!name) {
-      return suggestion;
-    }
-
-    return {
-      ...suggestion,
-      helpText: getHelpText(name, database, reportTimezone),
-    };
-  });
-
   return { suggestions };
 }
```

After applying this patch you'll notice that TS complains about `database` being possibly `null` which `getHelpText` does not accept. But `getHelpText` uses `database` only to pass it to `description` function which actually never uses `database` or `reportTimezone`. It's a good opportunity to remove some dead code.

Reply:
`description` unfortunately does use `database` and `reportTimezone` in [`getDescriptionForNow`](https://github.com/metabase/metabase/blob/f016befaefc930fab198ce57c05938b3c269a321/frontend/src/metabase-lib/v1/expressions/helper-text-strings.ts#L12)

Reply:
Oops, I missed it.
In that case we can call `getHelpText` conditionally.
Anyway, not a huge deal. Feel free to ignore my suggestion.